lib/splitclient-rb/clients/split_client.rb

```diff
@@ -1,114 +1,131 @@
+      bucketing_key, matching_key = keys_from_key(key)
+      bucketing_key = matching_key if bucketing_key.nil?
 
-  def get_treatments(key, split_names, attributes = nil)
-    bucketing_key, matching_key = keys_from_key(key)
-    bucketing_key = matching_key if bucketing_key.nil?
+      treatments_with_labels =
+        @splits_repository.get_splits(split_names).each_with_object({}) do |(name, data), memo|
+          memo.merge!(name => get_treatment(key, name, attributes, data, false, true))
+        end
 
-    treatments =
-      @splits_repository.get_splits(split_names).each_with_object({}) do |(name, data), memo|
-        memo.merge!(name => get_treatment(key, name, attributes, data, false))
+      if @config.impressions_queue_size > 0
+        @impressions_repository.add_bulk(matching_key, bucketing_key, treatments_with_labels, (Time.now.to_f * 1000.0).to_i)
       end
 
-    if @config.impressions_queue_size > 0
-      @impressions_repository.add_bulk(matching_key, bucketing_key, treatments, (Time.now.to_f * 1000.0).to_i)
+      split_names = treatments_with_labels.keys
+      treatments = treatments_with_labels.values.map { |v| v[:treatment] }
+
+      Hash[split_names.zip(treatments)]
     end
 
-    treatments
-  end
+    #
+    # obtains the treatment for a given feature
+    #
+    # @param key [String/Hash] user id or hash with matching_key/bucketing_key
+    # @param split_name [String/Array] name of the feature that is being validated or array of them
+    #
+    # @return [String/Hash] Treatment as String or Hash of treatments in case of array of features
+    def get_treatment(key, split_name, attributes = nil, split_data = nil, store_impressions = true, multiple = false)
+      bucketing_key, matching_key = keys_from_key(key)
+      bucketing_key = matching_key if bucketing_key.nil?
+
+      if matching_key.nil?
+        @config.logger.warn('matching_key was null for split_name: ' + split_name.to_s)
+        return parsed_treatment(multiple, { label: 'matching_key was null', treatment: Treatments::CONTROL })
+      end
 
-  #
-  # obtains the treatment for a given feature
-  #
-  # @param key [String/Hash] user id or hash with matching_key/bucketing_key
-  # @param split_name [String/Array] name of the feature that is being validated or array of them
-  #
-  # @return [String/Hash] Treatment as String or Hash of treatments in case of array of features
-  def get_treatment(key, split_name, attributes = nil, split_data = nil, store_impressions = true)
-    bucketing_key, matching_key = keys_from_key(key)
-    bucketing_key = matching_key if bucketing_key.nil?
-
-    if matching_key.nil?
-      @config.logger.warn('matching_key was null for split_name: ' + split_name.to_s)
-      return Treatments::CONTROL
-    end
+      if split_name.nil?
+        @config.logger.warn('split_name was null for key: ' + key)
+        return parsed_treatment(multiple, { label: 'split_name was null', treatment: Treatments::CONTROL })
```

Comment:
when the split_name is nil, there is not impression right? what happens in line 57 then?

` 'split_name was null'` is not part of the spec.